vscode/src/localiser.ts

```diff
@@ -4,42 +4,35 @@
 import * as l10nLib from '@vscode/l10n'
 
 import * as vscode from 'vscode';
+import { ORACLE_VSCODE_EXTENSION_ID } from './constants';
 
+const DEFAULT_BUNDLE_FILE = 'l10n/bundle.l10n.en.json';
 
 type TranslatorFn = typeof vscode.l10n.t
 
-export interface l10n{
-  getLocaleValue(key:string):string
-  getLocaleValueWithPlaceHolders(key: string, placeholderMap:Record<string, any>): string 
+export interface l10n {
+  value(key: string, placeholderMap?: Record<string, any>): string
 }
 
-class l10Wrapper implements l10n{
+class l10Wrapper implements l10n {
   private defaultTranslation: TranslatorFn;
-  constructor(extensionId:string,defaultBundlePath:string){
-    let defaultBundleAbsoluteFsPath = vscode.Uri.file(`${vscode.extensions.getExtension(extensionId)?.extensionPath}/${defaultBundlePath}`).fsPath
-    l10nLib.config({  
-          fsPath:defaultBundleAbsoluteFsPath
-    });
-    this.defaultTranslation = l10nLib.t;
-    
-  }
-  getLocaleValue(key: string): string {
-    if(vscode.l10n.bundle&&vscode.l10n.t(key)!==key){
-       return  vscode.l10n.t(key);
-    }else{
-      return this.defaultTranslation(key);
+    constructor(extensionId: string, defaultBundlePath: string) {
+      let defaultBundleAbsoluteFsPath = vscode.Uri.file(`${vscode.extensions.getExtension(extensionId)?.extensionPath}/${defaultBundlePath}`).fsPath
+      l10nLib.config({
+        fsPath: defaultBundleAbsoluteFsPath
+      });
+      this.defaultTranslation = l10nLib.t;
     }
-  }
 
-
-  getLocaleValueWithPlaceHolders(key: string, placeholderMap:Record<string, any>): string {
-    if(vscode.l10n.bundle&&vscode.l10n.t(key)!==key){
-       return  vscode.l10n.t(key,placeholderMap);
-    }else{
-      return this.defaultTranslation(key,placeholderMap);
+    value(key: string, placeholderMap: Record<string, any>): string {
+      const isPresentInBundle = vscode.l10n.bundle && vscode.l10n.t(key) !== key;
+      if (isPresentInBundle) {
+        return vscode.l10n.t(key, placeholderMap);
```

Comment:
This reads better. But, the optimisation can still be done by reusing the first returned value.
```
const valueFromBundle:string = vscode.l10n.bundle ? vscode.l10n.t(key, placeholderMap) : key;
const isPresentInBundle = valueFromBundle !== key;
return isPresentInBundle ? valueFromBundle : this.defaultTranslation(key, placeholderMap);
```

Reply:
first way looks clean, when you said cache i thought resource bundle cache got it thanks.